packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -428,4 +431,22 @@ class FlutterCommandRunner extends CommandRunner {
       })
       .toList();
   }
+
+  /// Check that the Flutter being run is the one we're expecting.
+  void _checkFlutterCopy() {
+    String cwd = path.normalize(path.absolute(Directory.current.path));
+
```

Comment:
in the future we should probably augment this to walk up the directory tree checking to see if we're in a subdirectory of a flutter checkout, too. That would catch, e.g., someone going into the examples directory and running a demo.